File: leetcode_buggy_version/java_bugg/Roman_To_Integer.java
package leetcode_buggy_version.java_bugg;

// https://leetcode.com/problems/roman-to-integer/solutions/5045094/roman-to-integer-switch-case-basic-approach/
class Roman_To_Integer {
  public int romanToInt(String s) {

    int sum = 0;

    for (int i = 0; i < s.length() - 1; i++) {

      switch (s.charAt(i)) {
        case ('I'):
          if (s.charAt(i + 1) == 'V' || s.charAt(i + 1) == 'X') {
            sum -= 1;
          } else {
            sum += 1;
          }
          break;

        case ('V'):
          sum += 5;
          break;

        case ('X'):
          if (s.charAt(i + 1) == 'L' || s.charAt(i + 1) == 'C') {
            sum -= 10;
          } else {
            sum += 10;
          }
          break;

        case ('L'):
          sum += 50;
          break;

        case ('C'):
          if (s.charAt(i + 1) == 'D' || s.charAt(i + 1) == 'M') {
            sum -= 100;
          } else {
            sum += 100;
          }
          break;

        case ('D'):
          sum += 500;
          break;

        case ('M'):
          sum += 1000;
          break;
      }
    }

    switch (s.charAt(s.length() - 1)) {
      case ('I'):
        sum += 1;
        break;

      case ('V'):
        sum += 5;
        break;

      case ('X'):
        sum += 10;
        break;

      case ('L'):
        sum += 50;
        break;

      case ('C'):
        sum += 100;
        break;

      case ('D'):
        sum += 500;
        break;

      case ('M'):
        sum += 1000;
        break;
    }

    return sum;

  }
}

File: leetcode_buggy_version/python/Roman_To_Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        number = 0
        temp = []
        n = len(s)
        value = 0
        before_value = 0
        for i in s[::-1]:
            if i == 'I':
                temp.append(1)
            elif i == 'V':
                temp.append(2)
            elif i == 'X':
                temp.append(3)
            elif i == 'L':
                temp.append(4)
            elif i == 'C':
                temp.append(5)
            elif i == 'D':
                temp.append(6)
            elif i == 'M':
                temp.append(7)
                
        for j in range(0, n):
            if temp[j] == 1:
                value = 1
            elif temp[j] == 2:
                value = 5
            elif temp[j] == 3:
                value = 10 
            elif temp[j] == 4:
                value = 50 
            elif temp[j] == 5:
                value = 100 
            elif temp[j] == 6:
                value = 500 
            elif temp[j] == 7:
                value = 1000

            if before_value > temp[j]:
                number += value
                before_value = temp[j]
            elif before_value < temp[j]:
                number += value
                before_value = temp[j]
            elif before_value == temp[j]:
                number += value
                before_value = temp[j]
                  
        return number
    

#https://leetcode.com/problems/roman-to-integer/solutions/5070487/roman-numerials-os/
